src/lib.rs

```diff
@@ -310,49 +310,43 @@
 
 #![deny(non_camel_case_types)]
 
-extern crate url;
+#[macro_use]
+extern crate combine;
 extern crate sha1;
+extern crate url;
 
-#[cfg(feature="with-rustc-json")]
-pub extern crate rustc_serialize as serialize;
-#[cfg(feature="with-unix-sockets")]
+#[macro_use]
+extern crate futures;
+extern crate tokio_core;
+#[macro_use]
+extern crate tokio_io;
+
+#[cfg(feature = "with-unix-sockets")]
 extern crate unix_socket;
+#[cfg(feature = "with-unix-sockets")]
+extern crate tokio_uds;
+#[cfg(feature = "with-rustc-json")]
+pub extern crate rustc_serialize as serialize;
 
 #[doc(hidden)]
-#[cfg(feature="with-rustc-json")]
+#[cfg(feature = "with-rustc-json")]
 pub use serialize::json::Json;
 
 // public api
-pub use parser::{parse_redis_value, Parser};
+pub use parser::{parse_redis_value, parse_async, Parser};
 pub use client::Client;
 pub use script::{Script, ScriptInvocation};
-pub use connection::{Connection, ConnectionLike, ConnectionInfo, ConnectionAddr,
-                     IntoConnectionInfo, PubSub, Msg, transaction, parse_redis_url};
-pub use cmd::{cmd, Cmd, pipe, Pipeline, Iter, pack_command};
+pub use connection::{parse_redis_url, transaction, Connection, ConnectionAddr, ConnectionInfo,
+                     ConnectionLike, IntoConnectionInfo, Msg, PubSub};
+pub use cmd::{cmd, pack_command, pipe, Cmd, Iter, Pipeline};
 pub use commands::{Commands, PipelineCommands};
 
-pub use types::{
-    /* low level values */
-    Value,
-
-    /* error and result types */
-    RedisError,
-    RedisResult,
-
-    /* error kinds */
-    ErrorKind,
-
-    /* utility types */
-    InfoDict,
-    NumericBehavior,
-
-    /* conversion traits */
-    FromRedisValue,
-    ToRedisArgs,
+pub use types::{/* utility functions */ from_redis_value, /* error kinds */ ErrorKind,
+                /* conversion traits */ FromRedisValue, /* utility types */ InfoDict,
+                NumericBehavior, /* error and result types */ RedisError, RedisResult,
+                ToRedisArgs, /* low level values */ Value, RedisFuture};
```

Comment:
This auto-formatted change makes it both harder to read and to change.
IMO we should keep the old formatting and mark it so rustfmt does not touch it.

Reply:
Fixed